src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -83,17 +83,49 @@ public static DateOnly FromDayNumber(int dayNumber)
         /// <summary>
         /// Gets the year component of the date represented by this instance.
         /// </summary>
-        public int Year => GetEquivalentDateTime().Year;
+        public int Year
+        {
+            get
+            {
+                // y100 = number of whole 100-year periods since 1/1/0001
+                // r1 = (day number within 100-year period) * 4
+                (uint y100, uint r1) = Math.DivRem((((uint)_dayNumber << 2) | 3U), DateTime.DaysPer400Years);
+                return 1 + (int)(100 * y100 + (r1 | 3) / DateTime.DaysPer4Years);
+            }
+        }
 
         /// <summary>
         /// Gets the month component of the date represented by this instance.
         /// </summary>
-        public int Month => GetEquivalentDateTime().Month;
+        public int Month
+        {
+            get
+            {
+                // r1 = (day number within 100-year period) * 4
+                uint r1 = ((((uint)_dayNumber << 2) | 3U) + 1224) % DateTime.DaysPer400Years;
+                ulong u2 = Math.BigMul(DateTime.EafMultiplier, r1 | 3U);
+                ushort daySinceMarch1 = (ushort)((uint)u2 / DateTime.EafDivider);
+                int n3 = 2141 * daySinceMarch1 + 197913;
+                return (ushort)(n3 >> 16) - (daySinceMarch1 >= DateTime.March1BasedDayOfNewYear ? 12 : 0);
+            }
+        }
 
         /// <summary>
         /// Gets the day component of the date represented by this instance.
         /// </summary>
-        public int Day => GetEquivalentDateTime().Day;
+        public int Day
+        {
+            get
+            {
+                // r1 = (day number within 100-year period) * 4
+                uint r1 = ((((uint)_dayNumber << 2) | 3U) + 1224) % DateTime.DaysPer400Years;
+                ulong u2 = Math.BigMul(DateTime.EafMultiplier, r1 | 3U);
+                ushort daySinceMarch1 = (ushort)((uint)u2 / DateTime.EafDivider);
+                int n3 = 2141 * daySinceMarch1 + 197913;
+                // Return 1-based day-of-month
+                return (ushort)n3 / 2141 + 1;
```

Comment:
here and in above - don't you just effectively inline DateTime.Day/Month/Year by hands?

Reply:
In effect it circumvents 64-bit multiplication, 64-bit bitwise-and, 64-bit division; and replaces it with a 32-bit shift.

Should we assume JIT takes care of that?

Reply:
I agree with @EgorBo. I don't think we need to complicate the code just to do what the JIT can do. I am concerned about the code maintainability. Keeping it simple is easier to touch in the future.